server/src/http/mod.rs

```diff
@@ -172,6 +186,7 @@ async fn handle_request(
 
 pub async fn run_http<F>(
     addr: SocketAddr,
+    authorizer: Arc<Mutex<Box<dyn Authorizer + Send + Sync>>>,
```

Comment:
looks like an `Arc<dyn Authorizer + Send + Sync>` would do the job here

Reply:
fixed